### PR TITLE
Refactors logging in `pkg/dashboard/ttlcontroller`

### PR DESCRIPTION
**Pull request by afzal442**
Signed-off-by: afzal442 <afzal442@gmail.com>

Closes: #2882 

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: pkg/dashboard/ttlcontroller/ttlcontroller.go [hunk 1]
```diff
@@ -53,7 +51,9 @@ func NewController(
 	event core.EventStore,
 	ttlc *TTLconfig,
 ) *Controller {
+	logger, _ := z.NewDefaultZapLogger()
```
Comment (STRRL) [inline pkg/dashboard/ttlcontroller/ttlcontroller.go:54]: please do not create another logger, acquire the logger as a parameter in the constructor.

**Review by STRRL — CHANGES_REQUESTED** [review]

**Comment by STRRL** [comment]
Hi @afzal442 do you need some help?

**Comment by STRRL** [comment]
Hi @afzal442 , here is a failed check called `ci / go (verify)`, it is related about code linters and code generation.

Please execute `make check` then commit the changes.

**Review by STRRL — CHANGES_REQUESTED** [review]

File: pkg/dashboard/ttlcontroller/ttlcontroller.go [hunk 2]
```diff
@@ -55,13 +52,15 @@ type TTLconfig struct {
...
 func NewController(
+	logger logr.Logger,
 	experiment core.ExperimentStore,
 	event core.EventStore,
 	schedule core.ScheduleStore,
 	workflow core.WorkflowStore,
 	ttlc *TTLconfig,
 ) *Controller {
 	return &Controller{
+		logger:     logger,
```
Comment (STRRL) [inline pkg/dashboard/ttlcontroller/ttlcontroller.go:63]: This constructor is used with the "dependency injection framework" called fx. We still need to give it the name "ttlcontroller".

So we could not just feed the "constructor" to the fx(cmd/chaos-dashboard/main.go:94), we need to create another function to do similar things, including naming. Please take a look at https://github.com/chaos-mesh/chaos-mesh/pull/3035, using the similar solution.

**Review by STRRL — CHANGES_REQUESTED** [review]

File: pkg/dashboard/ttlcontroller/fx.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+// limitations under the License.
+
+package ttlcontroller
+
+import (
+	"github.com/chaos-mesh/chaos-mesh/pkg/dashboard/core"
+	"github.com/go-logr/logr"
+)
+
+func InitInject(logger logr.Logger, experiment core.ExperimentStore, event core.EventStore, schedule core.ScheduleStore, workflow core.WorkflowStore, ttlc *TTLconfig) *Controller {
```
Comment (STRRL) [inline pkg/dashboard/ttlcontroller/fx.go:22]: There is a typo in the type of the last parameter, it should be `TTLConfig`.

And I prefer to use `Bootstrap` as the function name, with keeping the same style with other methods provided to `fx`

Rest LGTM!

Comment (afzal442) [reply pkg/dashboard/ttlcontroller/fx.go:22]: ` TTLconfig` seems to be correct here, I don't know why it's showing undefined.
😕

Comment (afzal442) [reply pkg/dashboard/ttlcontroller/fx.go:22]: > with keeping the same style with other methods provided to fx

do you mean I need to update the function to as below?
```go
func Bootstrap(conf *config.ChaosDashboardConfig, workflow core.WorkflowStore, logger logr.Logger) *Controller {
.
.
}
```

Comment (STRRL) [reply pkg/dashboard/ttlcontroller/fx.go]: > ` TTLconfig` seems to be correct here, I don't know why it's showing undefined. confused

It should be `TTLConfig` not `TTLconfig`, notice the capital character "C" in "Config"

Comment (STRRL) [reply pkg/dashboard/ttlcontroller/fx.go]: > > with keeping the same style with other methods provided to fx
> 
> do you mean I need to update the function to as below?
> 
> ```go
> func Bootstrap(conf *config.ChaosDashboardConfig, workflow core.WorkflowStore, logger logr.Logger) *Controller {
> .
> .
> }
> ```

Yes. I saw your commits, you did great. :D

**Comment by STRRL** [comment]
Hi @afzal442 , please run `make check` on your local machine, then commit the changes. :)

**Review by STRRL — APPROVED** [review]
LGTM!

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Pull request merged**